x-pack/plugins/siem/public/components/url_state/helpers.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { decode, encode, RisonValue } from 'rison-node';
+import { Location } from 'history';
+import { QueryString } from 'ui/utils/query_string';
+import { KqlQuery, LocationTypes } from './types';
+import { CONSTANTS } from './constants';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const decodeRisonUrlState = (value: string | undefined): RisonValue | any | undefined => {
+  try {
+    return value ? decode(value) : undefined;
+  } catch (error) {
+    if (error instanceof Error && error.message.startsWith('rison decoder error')) {
+      return {};
+    }
+    throw error;
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const encodeRisonUrlState = (state: any) => encode(state);
+
+export const getQueryStringFromLocation = (location: Location) => location.search.substring(1);
+
+export const getParamFromQueryString = (queryString: string, key: string): string | undefined => {
+  const queryParam = QueryString.decode(queryString)[key];
+  return Array.isArray(queryParam) ? queryParam[0] : queryParam;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const replaceStateKeyInQueryString = <UrlState extends any>(
+  stateKey: string,
+  urlState: UrlState | undefined
+) => (queryString: string) => {
+  const previousQueryValues = QueryString.decode(queryString);
+  const encodedUrlState =
+    typeof urlState !== 'undefined' ? encodeRisonUrlState(urlState) : undefined;
```

Comment:
I usually only see this type of check against `undefined` if we are worried about `undefined` being re-defined which isn't possible.

I would say to just do `urlState !== undefined`, but I know that this code was copied from another location.

Maybe what's best is to put a piece of documentation that says:

```ts
// Code was copied from x-pack/plugins/infra/public/utils/url_state.tsx
// Remove this if these utilities are promoted to kibana core 
```

In another part of the code someone used the word hard fork like so:

```ts
/** A hard-fork of the `infra` `AutoSizer` ಠ_ಠ */
```

So the  ಠ_ಠ face is acceptable as well to designate we are taking this code as-is and we have to from time to time check for bug fixes from the other locations.